service/src/test/java/uk/nhs/adaptors/gp2gp/common/service/RandomIdGeneratorServiceTest.java

```diff
@@ -13,11 +14,13 @@
     public void When_GeneratingRandomId_Expect_GeneratedIdIsRandomUUID() {
         String id1 = new RandomIdGeneratorService().createNewId();
         String id2 = new RandomIdGeneratorService().createNewId();
-        assertThatCode(() -> UUID.fromString(id1))
-            .doesNotThrowAnyException();
-        assertThatCode(() -> UUID.fromString(id2))
-            .doesNotThrowAnyException();
-        assertThat(id1).isNotEqualTo(id2);
-    }
 
+        assertAll(
+            () -> assertThatCode(() -> UUID.fromString(id1)).doesNotThrowAnyException(),
+            () -> assertThatCode(() -> UUID.fromString(id2)).doesNotThrowAnyException(),
+            () -> assertThat(id1).isNotEqualTo(id2),
+            () -> assertThat(id1).matches("[A-Z0-9-]+"),
+            () -> assertThat(id2).matches("[A-Z0-9-]+")
```

Comment:
Regex for UUIDs is `[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}`
For uppercase we simply this to use `[0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12}` to test against?